firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -54,6 +54,10 @@
  * helper routines that make dealing with SQLite much more pleasant.
  */
 public final class SQLitePersistence extends Persistence {
+  enum DataMigration {
```

Comment:
FWIW, the Android Style guide says to steer away from Enums due to their overhead.

Reply:
The alternative is String constants, which is also not ideal. I'd like to keep it, the space overhead here is acceptable IMO.

Reply:
I personally agree but I don't want to lose my job over violating the Android Style Guide:

g3doc/company/teams/android-api-council/guidelines/index.md?cl=head#avoid-enum

You can see me discuss this with Gil when you look at the history for https://g3doc.corp.google.com/company/teams/android-api-council/guidelines/index.md?cl=head#avoid-enum

Note that this is not necessarily only about space overhead, but also about limiting the number of symbols that have to be loaded as Android has a pretty strict limit for the number of symbols a single DEX can contain. While we are already well beyond the limit for a single DEX file, it still benefits our loading speed if we try to limit the number of symbols.

That being said, Enums seem to be the only place where Android developers limit the number of methods.

Reply:
Fair, switched to string constant.